src/execution/compiler/codegen.cpp

```diff
@@ -116,33 +122,43 @@ ast::Expr *CodeGen::TableIterInit(ast::Identifier tvi, uint32_t table_oid, ast::
 
 ast::Expr *CodeGen::PCIGet(ast::Identifier pci, type::TypeId type, bool nullable, uint32_t idx) {
   ast::Builtin builtin;
+  ast::Type *ast_type;
   switch (type) {
     case type::TypeId::BOOLEAN:
       builtin = nullable ? ast::Builtin::PCIGetBoolNull : ast::Builtin::PCIGetBool;
+      ast_type = ast::BuiltinType::Get(Context(), ast::BuiltinType::Bool);
```

Comment:
1. This should be `ast::BuiltinType::Boolean`. Yes, its confusing.
2. While the change is necessary, this should be done in all the functions in `codegen.h`. I'd hold off on it for now, and include this in codegen V2 when it gets ported over.

Reply:
Added a TODO.